lib/ruby_indexer/lib/ruby_indexer/rbs_indexer.rb

```diff
@@ -122,7 +127,106 @@ def handle_method(member, owner)
       end
 
       real_owner = member.singleton? ? @index.existing_or_new_singleton_class(owner.name) : owner
-      @index.add(Entry::Method.new(name, file_path, location, location, comments, [], visibility, real_owner))
+      signatures = signatures(member)
+      @index.add(Entry::Method.new(name, file_path, location, location, comments, signatures, visibility, real_owner))
+    end
+
+    sig { params(member: RBS::AST::Members::MethodDefinition).returns(T::Array[Entry::Signature]) }
+    def signatures(member)
+      member.overloads.map do |overload|
+        parameters = process_overload(overload)
+        Entry::Signature.new(parameters)
+      end
+    end
+
+    sig { params(overload: RBS::AST::Members::MethodDefinition::Overload).returns(T::Array[Entry::Parameter]) }
+    def process_overload(overload)
+      function = T.cast(overload.method_type.type, RBS::Types::Function)
+      parameters = parse_arguments(function)
+
+      block = overload.method_type.block
+      parameters << Entry::BlockParameter.anonymous if block&.required
+
+      parameters
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::Parameter]) }
+    def parse_arguments(function)
+      parameters = []
+      parameters.concat(process_required_and_optional_positionals(function))
+      parameters.concat(process_trailing_positionals(function)) if function.trailing_positionals
+      parameters << process_rest_positionals(function) if function.rest_positionals
+      parameters.concat(process_required_keywords(function)) if function.required_keywords
+      parameters.concat(process_optional_keywords(function)) if function.optional_keywords
+      parameters << process_rest_keywords(function) if function.rest_keywords
+      parameters
+    end
+
+    sig { params(function: RBS::Types::Function).returns(T::Array[Entry::RequiredParameter]) }
+    def process_required_and_optional_positionals(function)
```

Comment:
This was prevously two methods, I combined them because we need a way to keep track of an index to use for `arg0` etc, and this seemed the simplest way.